src/_pytest/main.py

```diff
@@ -367,6 +365,47 @@ def pytest_runtestloop(session: Session) -> bool:
     return True
 
 
+def _is_setuptools_in_pyproject_toml(toml: Path) -> bool:
+    """Attempt to decode a toml file into a dict, returning None if fails."""
+    if sys.version_info >= (3, 11):
+        import tomllib
+    else:
+        import tomli as tomllib
+
+    try:
+        toml_text = toml.read_text(encoding="utf-8")
+        parsed_toml = tomllib.loads(toml_text)
+        build_system = parsed_toml.get("build-system", {}).get("requires")
+        if "setuptools" in build_system:
+            return True
+    except Exception:
```

Comment:
I would prefer to avoid using a `except Exception` here unless absolute necessary -- from what I see on the code inside the try/except block, seems that we are taking in account missing keys already.

Did you account that `tomlib.loads` might fail due to a malformed TOML file? If so let's catch the precise exception here instead.